src/main/scala/com/fulcrumgenomics/umi/VanillaUmiConsensusCaller.scala

```diff
@@ -246,7 +246,7 @@ class VanillaUmiConsensusCaller(override val readNamePrefix: String,
     }
     else {
       val sourceRecords   = records.flatMap(toSourceRead(_, this.options.minInputBaseQuality, this.options.qualityTrim))
-      val filteredRecords = filterToMostCommonAlignment(sourceRecords)
+      val filteredRecords = filterToMostCommonAlignment(sourceRecords).sortBy { rec => sourceRecords.indexOf(rec) }
```

Comment:
I'd also be open to not requiring a sort, by instead returning a list of the _index_ of the read in the original list for those that are kept.